### PR TITLE
Add pinning propogation to the mamba checks.

### PR DESCRIPTION
**Pull request by mariusvniekerk**
This will ensure that we don't issue a PR when not all python versions
have a valid run environment

### CODE REVIEW

File: conda_forge_tick/mamba_solver.py [hunk 1]
```diff
@@ -280,7 +280,7 @@ def is_recipe_solvable(feedstock_dir):
             arch,
         )
         solvable = solvable and _solvable
-        errors.extend(_errors)
+        errors.extend(["f{cbc_fname}: {e}" for e in _errors])
```
Comment (beckermr) [inline conda_forge_tick/mamba_solver.py:283]: I already put a fix in for this. Can you merge master in and figure out which one we should keep. I think yours might be better.

Comment (beckermr) [reply conda_forge_tick/mamba_solver.py:283]: Also you need to extract the basename here. This is a full path.

Comment (mariusvniekerk) [reply conda_forge_tick/mamba_solver.py]: made a hybrid

**Review by beckermr — APPROVED** [review]

**Comment by beckermr** [comment]
What?

> conda.exceptions.InvalidVersionSpec: Invalid version '>=4.0': invalid character(s)

**Comment by beckermr** [comment]
Once the tests are fixed we can merge @mariusvniekerk

**Comment by beckermr** [comment]
Does the latest version still handle python properly? Maybe we should add a test?

File: conda_forge_tick/mamba_solver.py [hunk 2]
```diff
@@ -217,7 +218,7 @@ def _mamba_factory(channels, platform):
     return MambaSolver(list(channels), platform)
 
 
-def is_recipe_solvable(feedstock_dir):
+def is_recipe_solvable(feedstock_dir) -> Tuple[bool, List[str], Dict[str, bool]]:
```
Comment (beckermr) [inline conda_forge_tick/mamba_solver.py:221]: you need to update the doc string.

File: conda_forge_tick/mamba_solver.py [hunk 3]
```diff
@@ -282,8 +284,9 @@ def is_recipe_solvable(feedstock_dir):
         solvable = solvable and _solvable
         cbc_name = os.path.basename(cbc_fname).rsplit(".", maxsplit=1)[0]
         errors.extend([f"{cbc_name}: {e}" for e in _errors])
+        solvable_by_cbc[cbc_name] = _solvable
 
-    return solvable, errors
+    return solvable, errors, solvable_by_cbc
```
Comment (beckermr) [inline conda_forge_tick/mamba_solver.py:289]: There are two branches above that return early. They need a third return argument too.

**Review by beckermr — CHANGES_REQUESTED** [review]

**Pull request merged**